File: lecture-5/isSorted2.cpp
#include <iostream>
using namespace std;

bool isSorted2(int arr[], int length) {
  for(int i = 0; i < (length - 1); i++) {
    if(arr[i] > arr[i + 1]) { return false; }
  }
  return true;
}
int main() {
  int LENGTH = 4500000;
  int a[LENGTH];
  for(int i = 0; i < LENGTH; i++) {
    a[i] = i;
  }
  cout << "Should be true: " << isSorted2(a, LENGTH) << endl;
}

File: lecture3.1/isSorted2.py

def is_sorted2(l):
  for i in range(0, len(l) - 1):
    if l[i] > l[i + 1]:
      return False
  return True

LENGTH = 1000000
l = list(range(0, LENGTH))
print "Length is: %s" % LENGTH
print is_sorted2(l)
